src/expr/src/scalar/mod.rs

```diff
@@ -742,6 +742,53 @@ impl MirScalarExpr {
                         {
                             // Canonically order elements so that deduplication works better.
                             mem::swap(expr1, expr2);
+                        } else if let (
+                            BinaryFunc::Eq,
+                            MirScalarExpr::Literal(
+                                Ok(lit_row),
+                                ColumnType {
+                                    scalar_type:
+                                        ScalarType::Record {
+                                            fields: field_types,
+                                            ..
+                                        },
+                                    ..
+                                },
+                            ),
+                            MirScalarExpr::CallVariadic {
+                                func: VariadicFunc::RecordCreate { .. },
+                                exprs: rec_create_args,
+                            },
+                        ) = (func, (**expr1).clone(), (**expr2).clone())
+                        {
+                            // Literal(record(c1, c2)) = record_create(e1, e2)
+                            //  -->
+                            // c1 = e1 AND c2 = e2
+                            //
+                            // `literal_constraints` relies on this transform, because
+                            // `(e1,e2) IN ((1,2))` is desugared using `record_create`.
+
+                            match lit_row.unpack_first() {
+                                Datum::List(datum_list) => {
+                                    *e = MirScalarExpr::CallVariadic {
+                                        func: VariadicFunc::And,
+                                        exprs: datum_list
+                                            .iter()
+                                            .zip(field_types)
+                                            .zip(rec_create_args)
+                                            .map(|((d, (_, typ)), a)| MirScalarExpr::CallBinary {
+                                                func: BinaryFunc::Eq,
+                                                expr1: Box::new(MirScalarExpr::Literal(
```

Comment:
nit: you can use `MirScalarExpr::literal_ok(d, typ)` here.

Reply:
It wouldn't be more convenient, because `literal_ok` expects a `ScalarType`, but I have a `ColumnType` in my hand here.